src/torchmetrics/utilities/plot.py

```diff
@@ -0,0 +1,264 @@
+                xlim[0], upper_bound, s="Optimal \n value", horizontalalignment="center", verticalalignment="center"
+            )
+    return fig, ax
+
+
+def _get_col_row_split(n: int) -> Tuple[int, int]:
+    """Split n curves into rows x cols figures."""
+    nsq = sqrt(n)
+    if nsq * nsq == n:
+        return nsq, nsq
+    elif floor(nsq) * ceil(nsq) > n:
+        return floor(nsq), ceil(nsq)
+    else:
+        return ceil(nsq), ceil(nsq)
+
+
+def trim_axs(axs: Union[matplotlib.axes.Axes, np.ndarray], N: int) -> np.ndarray:
+    """Reduce *axs* to *N* Axes.
+
+    All further Axes are removed from the figure.
+    """
+    if isinstance(axs, matplotlib.axes.Axes):
+        return axs
+    else:
+        axs = axs.flat
+        for ax in axs[N:]:
+            ax.remove()
+        return axs[:N]
+
+
+def plot_confusion_matrix(
+    confmat: Tensor,
+    add_text: bool = True,
+    labels: Optional[List[str]] = None,
+    cmap: Optional[matplotlib.colors.Colormap] = None,
+) -> Tuple[plt.Figure, Union[matplotlib.axes.Axes, np.ndarray]]:
+    """Inspired by: https://github.com/scikit-learn/scikit-
+    learn/blob/main/sklearn/metrics/_plot/confusion_matrix.py."""
+
+    _error_on_missing_matplotlib()
+
+    if confmat.ndim == 3:  # multilabel
+        n, n_classes = confmat.shape[0], 2
+        rows, cols = _get_col_row_split(n)
+    else:
+        n, n_classes, rows, cols = 1, confmat.shape[0], 1, 1
+
+    if labels is not None and confmat.ndim != 3 and len(labels) != n_classes:
+        raise ValueError(
+            "Expected number of elements in arg `labels` to match number of labels in confmat but "
+            f"got {len(labels)} and {n_classes}"
+        )
+    labels = labels if labels is not None else list(range(n_classes))
+
+    fig, axs = plt.subplots(nrows=rows, ncols=cols)
+    axs = trim_axs(axs, n)
+    for i in range(n):
+        if rows != 1 and cols != 1:
+            ax = axs[i]
+            ax.set_title(f"Label {i}", fontsize=15)
+        else:
+            ax = axs
+        ax.imshow(confmat[i].cpu().detach() if confmat.ndim == 3 else confmat.cpu().detach(), cmap=cmap)
+        ax.set_xlabel("True class", fontsize=15)
+        ax.set_ylabel("Predicted class", fontsize=15)
+        ax.set_xticks(list(range(n_classes)))
+        ax.set_yticks(list(range(n_classes)))
+        ax.set_xticklabels(labels, rotation=45, fontsize=10)
+        ax.set_yticklabels(labels, rotation=25, fontsize=10)
+
+        if add_text:
+            for ii, jj in product(range(n_classes), range(n_classes)):
+                val = confmat[i, ii, jj] if confmat.ndim == 3 else confmat[ii, jj]
+                ax.text(jj, ii, str(val.item()), ha="center", va="center", fontsize=15)
+
+    return fig, axs
+
+
+def _plot_curve(
+    input: Union[Tuple[Tensor, Tensor, Tensor], Tuple[List[Tensor], List[Tensor], List[Tensor]]],
+    auc: bool = False,
+    single_plot: bool = False,
+    xy_labels: Tuple[str, str] = ("X axis", "Y axis"),
```

Comment:
also support to pass axes? What if I by choice want multiple plots on the same axes?

Reply:
will update